### PR TITLE
ERR: Raise ValueError when week is passed in to_datetime format witho…

### LINKED ISSUE #16774: to_datetime parses dates incorrectly when the format includes '%W' and does not include day of week plus calendar year
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

# I'm not sure how a bare week should be interpreted.  But the result should
# probably still be of the same week number
print(pd.to_datetime('20', format='%W').strftime('%W'))  # 01

# Parsing is broken even when the week is fully specified
print(pd.to_datetime('2017-20', format='%Y-%W').strftime('%Y-%W'))  # 2017-00
```
#### Problem description

to_datetime parses a week XX as 0 for any valid XX.  Interestingly, an impossible XX (ex: '70') throws a ValueError as expected.

#### Expected Output
20
2017-20

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.4.3.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-79-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.20.2
pytest: None
pip: 9.0.1
setuptools: 36.0.1
Cython: 0.24.1
numpy: 1.13.0
scipy: 0.18.1
xarray: None
IPython: 6.1.0
sphinx: None
patsy: None
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.0.0
openpyxl: 2.4.0
xlrd: 1.0.0
xlwt: None
xlsxwriter: None
lxml: 3.3.3
bs4: 4.2.1
html5lib: 0.999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by reidy-p**
…ut day or year (#16774)

- [x] closes #16774
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexes/datetimes/test_tools.py [hunk 1]
```diff
@@ -372,6 +372,19 @@ def test_datetime_invalid_datatype(self):
...
+                             [('2017-20', '%Y-%W'),
+                              ('20 Sunday', '%W %A'),
+                              ('20 Sun', '%W %a'),
+                              ('2017-21', '%Y-%U'),
+                              ('20 Sunday', '%U %A'),
+                              ('20 Sun', '%U %a')])
+    def test_week_without_day_and_calendar_year(self, date, format):
+        # GH16774
+
+        with pytest.raises(ValueError):
```
Comment (gfyoung) [inline pandas/tests/indexes/datetimes/test_tools.py]: Instead of `pytest.raises`, use `tm.assert_raises_regex` to check for the error message as well.

Comment (reidy-p) [reply pandas/tests/indexes/datetimes/test_tools.py]: Thanks. Done.

File: pandas/core/tools/datetimes.py [hunk 2]
```diff
@@ -254,6 +254,16 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
                 require_iso8601 = not infer_datetime_format
                 format = None
 
+        if format is not None:
```
Comment (jreback) [inline pandas/core/tools/datetimes.py]: can you move this check to

``pandas/_libs/tslibs/strptime.pyx``

its just closer to the code where we use it

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. pls move where we do this check. ping on green.

**Comment by jreback** [comment]
lgtm ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @reidy-p

**Pull request merged**